packages/core/src/server/resolver.ts

```diff
@@ -290,10 +290,33 @@ const authorize: ResolverAuthorize = (...args) => {
   }
 }
 
+export type ParserType = "sync" | "async"
+
+function zod<Schema extends ZodSchema<any, any>, Type = zInfer<Schema>>(
+  schema: Schema,
+  parserType: "sync",
+): (input: Type) => Type
+function zod<Schema extends ZodSchema<any, any>, Type = zInfer<Schema>>(
+  schema: Schema,
+  parserType: "async",
+): (input: Type) => Promise<Type>
+function zod<Schema extends ZodSchema<any, any>, Type = zInfer<Schema>>(
+  schema: Schema,
+): (input: Type) => Promise<Type>
+function zod<Schema extends ZodSchema<any, any>, Type = zInfer<Schema>>(
+  schema: Schema,
+  parserType: ParserType = "async",
+){
+  if(parserType === "sync"){
+    return (input: Type) : Type => schema.parse(input)
+  }else{
+    return async (input: Type) : Promise<Type> => schema.parseAsync(input)
```

Comment:
Don't have to mark as async since we don't use `await` inside the body
```suggestion
    return (input: Type) : Promise<Type> => schema.parseAsync(input)
```